### PR TITLE
conf matrix metric and confidence

### PR DESCRIPTION
**Pull request by msokoloff1**

### CODE REVIEW

File: labelbox/data/annotation_types/metrics/base.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from pydantic.error_wrappers import ErrorWrapper
+from typing import Dict, Optional, Any, Union
+
+ConfidenceValue = confloat(ge=0, le=1)
+
+MIN_CONFIDENCE_SCORES = 2
+MAX_CONFIDENCE_SCORES = 15
+
+
+class BaseMetric(BaseModel, ABC):
```
Comment (tomislav-peharda) [inline labelbox/data/annotation_types/metrics/base.py:12]: `ABC`?

File: labelbox/data/annotation_types/metrics/base.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from abc import ABC
+from pydantic import ValidationError, confloat, BaseModel, validator
+from pydantic.error_wrappers import ErrorWrapper
+from typing import Dict, Optional, Any, Union
+
+ConfidenceValue = confloat(ge=0, le=1)
+
+MIN_CONFIDENCE_SCORES = 2
+MAX_CONFIDENCE_SCORES = 15
```
Comment (tomislav-peharda) [inline labelbox/data/annotation_types/metrics/base.py:9]: how come `[2,15]` and not `[0,1]`?

**Review by tomislav-peharda — APPROVED** [review]
I'm giving it a stamp, but would be ideal if we could have someone else look at it too as I don't feel I'm good fit for this PR.

File: labelbox/data/annotation_types/metrics/base.py (hunk 2 above)
Comment (msokoloff1) [reply labelbox/data/annotation_types/metrics/base.py:9]: This is the number of scores. Not the range. We are constraining the range with pydantics confloat type. So there is no point of having confidence if you have less than one value. And if you have more than 15 it is too many details and clutters our db.

File: labelbox/data/annotation_types/metrics/base.py (hunk 1 above)
Comment (msokoloff1) [reply labelbox/data/annotation_types/metrics/base.py:12]: abstract base class. It prevents users from directly instantiating this object.

**Pull request merged**